src/components/documentManager.ts

```diff
@@ -129,65 +132,65 @@ export abstract class DocumentManager {
      * on the details page. This happens when no media is playing,
      * and the connected client is browsing the library.
      *
-     * @param {BaseItemDto} item to show information about
-     * @returns {Promise<void>} for the page to load
+     * @param item - to show information about
+     * @returns for the page to load
      */
-    public static showItem(item: BaseItemDto): Promise<void> {
+    public static async showItem(item: BaseItemDto): Promise<void> {
         // no showItem for cc audio
         if (getActiveDeviceId() === deviceIds.AUDIO) {
-            return Promise.resolve();
+            return;
         }
 
         // stop cycling backdrops
         this.clearBackdropInterval();
 
-        return Promise.all([
-            this.getWaitingBackdropUrl(item),
-            this.getPrimaryImageUrl(item),
-            this.getLogoUrl(item)
-        ]).then((urls) => {
-            requestAnimationFrame(() => {
-                this.setWaitingBackdrop(urls[0], item);
-                this.setDetailImage(urls[1]);
-                this.setLogo(urls[2]);
-
-                this.setOverview(item.Overview ?? null);
-                this.setGenres(item?.Genres?.join(' / ') ?? null);
-                this.setDisplayName(item);
-                this.setMiscInfo(item);
-
-                this.setRating(item);
-
-                if (item?.UserData?.Played) {
-                    this.setPlayedIndicator(true);
-                } else if (item?.UserData?.UnplayedItemCount) {
-                    this.setPlayedIndicator(item?.UserData?.UnplayedItemCount);
-                } else {
-                    this.setPlayedIndicator(false);
-                }
+        const urls = [
+            await this.getWaitingBackdropUrl(item),
+            await this.getPrimaryImageUrl(item),
+            await this.getLogoUrl(item)
+        ];
```

Comment:
I'd like to keep the same behavior here, I did have a shot at it in https://github.com/jellyfin/jellyfin-chromecast/pull/190/files#diff-f7d8d074ad9f2f33a0ea4c9774f0aca90b80d72e4e8dba4f79dcdfc9283228d6R144-R183

Basically we switch to async/await but keep Promise.all() around.

Reply:
It's network requests so doing 3 in parallell is much better than 3 in serial..

Reply:
Yey :)